config/state_machine/moj.yml

```diff
@@ -309,26 +321,134 @@ case_types:
+
 
           approver:
             states:
               unassigned:
                 accept_approver_assignment:
+                  if: Case::BasePolicy#can_accept_or_reject_approver_assignment?
                 add_message_to_case:
                   if: Case::BasePolicy#can_add_message_to_case?
                 flag_for_clearance:
+                link_a_case:
                 reassign_user:
                   if: Case::BasePolicy#assignments_reassign_user?
+                remove_linked_case:
+                take_on_for_approval:
+                  if: Case::BasePolicy#can_take_on_for_approval?
+                unaccept_approver_assignment:
+                  if: Case::BasePolicy#can_unaccept_approval_assignment?
+                  switch_workflow: standard
+                unflag_for_clearance:
+                  switch_workflow: standard
 
+              awaiting_responder:
+                accept_approver_assignment:
+                  if: Case::BasePolicy#can_accept_or_reject_approver_assignment?
+                add_message_to_case:
+                  if: Case::BasePolicy#can_add_message_to_case?
+                flag_for_clearance:
+                link_a_case:
+                reassign_user:
+                remove_linked_case:
+                take_on_for_approval:
+                  if: Case::BasePolicy#can_take_on_for_approval?
+                unaccept_approver_assignment:
+                  if: Case::BasePolicy#can_unaccept_approval_assignment?
+                  switch_workflow: standard
+                unflag_for_clearance:
+                  switch_workflow: standard
+
+              drafting:
+                accept_approver_assignment:
+                  if: Case::BasePolicy#can_accept_or_reject_approver_assignment?
+                add_message_to_case:
+                  if: Case::BasePolicy#can_add_message_to_case?
+                extend_for_pit:
+                  if: Predicates#user_is_assigned_disclosure_specialist?
+                flag_for_clearance:
+                link_a_case:
+                reassign_user:
+                remove_linked_case:
+                take_on_for_approval:
+                  if: Case::BasePolicy#can_take_on_for_approval?
+                unaccept_approver_assignment:
+                  if: Case::BasePolicy#can_unaccept_approval_assignment?
+                  switch_workflow: standard
+                unflag_for_clearance:
+                  switch_workflow: standard
 
           responder:
             states:
               unassigned:
                 add_message_to_case:
                   if: Case::BasePolicy#can_add_message_to_case?
 
+              awaiting_responder:
+                accept_responder_assignment:
+                  if: Predicates#responder_is_member_of_assigned_team?
+                  transition_to: drafting
+                add_message_to_case:
+                  if: Predicates#responder_is_member_of_assigned_team?
```

Comment:
Is this right? Every other `add_message_to_case` event uses `Case::BasePolicy#can_add_message_to_case?`.

Reply:
responders can only add a message to a case if they are in the responding team, so we need a predicate for that.  I don't think we need to all ```Case::BasePolicy#can_add_message_to_case?``` for the other roles because managers can always add a message, and we need another predicate to check that the approver is an approver on the case.  
Will fix.